File: single-language-analysis/dataset-files/python/test_smell_145.py
# Original URL: https://github.com/Ranack/twit_sentiments/blob/e0aca8c7d97b34dbd268a5295598b29296546d4b/test_api.py#L50-L66

def test_smell_145(headers):
    """Test avec un texte vide."""
    data = {"text": ""}

    # Envoi de la requête POST
    response = requests.post(BASE_URL, json=data, headers=headers)

    # Afficher la réponse brute pour le débogage
    print(f"Response: {response.text}")

    # Vérification du code de statut (on s'attend à une erreur 400 pour un texte vide)
    assert response.status_code == 400, f"Expected 400, got {response.status_code} - {response.text}"

    # Vérification du message d'erreur dans la réponse JSON
    response_json = response.json()
    assert "detail" in response_json, "Error response missing 'detail'"
    assert response_json["detail"] == "Le texte ne peut pas être vide.", f"Expected error message 'Le texte ne peut pas être vide.', got {response_json['detail']}"

File: single-language-analysis/dataset-files/java/TestSmell145.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.util.HashMap;
import java.util.Map;

public class TestSmell145 {

    @Test
    public void testPredictEmptyText() {
        Map<String, String> data = new HashMap<>();
        data.put("text", "");

        // Envoi de la requête POST
        Response response = requests.post(BASE_URL, data, headers);

        // Afficher la réponse brute pour le débogage
        System.out.println("Response: " + response.getText());

        // Vérification du code de statut (on s'attend à une erreur 400 pour un texte vide)
        assertEquals(400, response.getStatusCode());

        // Vérification du message d'erreur dans la réponse JSON
        Map<String, Object> responseJson = response.json();
        assertTrue(responseJson.containsKey("detail"));
        assertEquals("Le texte ne peut pas être vide.", responseJson.get("detail"));
    }
}